docs/controller.md

```diff
@@ -1,40 +1,358 @@
-    fetch_data_audit_log
+```ruby
+class ProductsController < Wallaby::ResourcesController
+  def edit
+    super do
+      # do something here before rendering
+    end
   end
 end
 ```
 
-`Wallaby::ResourcesController` inherits `ApplicationController`, which means that all helper methods from `ApplicationController` will be available for any custom actions.
+Basically, `edit` action is simple as:
 
-If `ProductsController` is already occupied for other purpose, we could use another name and specify the model class so that Wallaby could identify it:
+```ruby
+def edit
+  authorize! :edit, resource
+  yield if block_given? # after_edit
+  respond_with resource
+end
+```
+
+that it can be completely replaced, just need to bear in mind that `@resource` is needed in the view.
+
+### Update
+
+- To add functionality before the action, it is fine to either use `before_action` or:
 
 ```ruby
-#!app/controllers/admin/products_controller.rb
-class Admin::ProductsController < Wallaby::ResourcesController
-  def self.model_class
-    Product
+class ProductsController < Wallaby::ResourcesController
+  def update
+    # do something here before saving the record
+    # to access the record, use `resource`
+    resource
+    # to re-assign the resource, assign to `@resource`
+    @resource = Product.new(new_arrival: true)
+    super
+  end
+end
+```
+
+- To add functionality after the action but before rendering, it goes:
+
+```ruby
+class ProductsController < Wallaby::ResourcesController
+  def update
+    super do
+      # do something here before rendering
+    end
+  end
+end
+```
+
+Basically, `update` action is simple as:
+
+```ruby
+def update
+  authorize! :update, resource
+  current_model_service.update resource, params
+  yield if block_given? # after_update
+  respond_with resource, location: helpers.show_path(resource)
+end
+```
+
+that it can be completely replaced, just need to bear in mind that `@resource` is needed.
+
+### Destroy
+
+- To add functionality before the action, it is fine to either use `before_action` or:
+
+```ruby
+class ProductsController < Wallaby::ResourcesController
+  def destroy
+    # do something here before saving the record
+    # to access the record, use `resource`
+    resource
+    # to re-assign the resource, assign to `@resource`
+    @resource = Product.new(new_arrival: true)
+    super
+  end
+end
+```
+
+- To add functionality after the action but before rendering, it goes:
+
+```ruby
+class ProductsController < Wallaby::ResourcesController
+  def destroy
+    super do
+      # do something here before rendering
+    end
+  end
+end
+```
+
+Basically, `destroy` action is simple as:
+
+```ruby
+def destroy
+  authorize! :destroy, resource
+  current_model_service.destroy resource, params
+  yield if block_given? # after_destroy
+  respond_with resource, location: helpers.index_path(current_model_class)
+end
+```
+
+that it can be completely replaced with your own implementation.
+
+### resource_params
```

Comment:
How about `Strong Parameters`?

Reply:
can you be more specific?

Reply:
I think this section should be called `Strong Parameters` 🙂

```md
### Strong Parameters

etc...
```